math/kth/kthnumber_test.go

```diff
@@ -0,0 +1,91 @@
+package kth
+
+import (
+	"testing"
+)
+
+func TestFindKthMax(t *testing.T) {
+	sortTests := []struct {
+		input    []int
+		k        int
+		expected int
+		name     string
+	}{
+		{
+			input:    []int{6, 7, 0, -1, 10, 70, 8, 22, 3, 9},
+			k:        3,
+			expected: 10,
+			name:     "3th largest number",
+		},
```

Comment:
This edge case will fail (I think) since there is no difference between what is returned vs the supposed error value

```suggestion
		{
			input:    []int{6, 7, 0, -1, 10, 70, 8, 22, 3, 9},
			k:        3,
			expected: 10,
			name:     "3th largest number",
		},
		{
			input:    []int{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1},
			k:        3,
			expected: -1,
			name:     "3th largest number",
		},
		{
			input:    []int{-1, -1, -1, -1, -1, -1},
			k:        7,
			expected: -1,
			name:     "This should be an error",
		},
```